pallets/author-reward-dest/src/lib.rs

```diff
@@ -0,0 +1,181 @@
+//! Pallet for setting and getting author reward destination.
+//!
+//! The author of a block can provide an account using inherents, this account is meant to be used
+//! for reward within the block such as tips and part of transaction fees.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use primitives_author_reward_dest::{InherentError, InherentType, INHERENT_IDENTIFIER};
+
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// The type `Self::AccountId` but with additional constraint.
+		type AccountIdType: IsType<<Self as frame_system::Config>::AccountId> + From<InherentType>;
+	}
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_finalize(_: BlockNumberFor<T>) {
+			// ensure we never go to trie with these values.
+			<Author<T>>::take().expect("`set_author_reward_dest` inherent must be present");
+		}
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// The inherent set author reward dest has already been submitted.
+		AlreadySet,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Set the author reward destination for the block.
+		#[pallet::call_index(0)]
+		#[pallet::weight((
+			T::DbWeight::get().writes(1),
```

Comment:
fwiw this is not really POV friendly as types like `RuntimeDbWeight` and such don't track PoV at all. 

The solution for now is to not use this "manual" weighing in code and only write and run benchmarks. 

I can't find it, but we recently re-discovered this and made an issue to solve it or want users about it better cc @ggwpez

Reply:
or to provide a weight metered DB: https://github.com/polkadot-fellows/RFCs/issues/49

Reply:
But anyway we will do becnhmarks in a final sprint of the runtime